jlog.c

```diff
@@ -2179,6 +2179,139 @@ int jlog_ctx_read_message(jlog_ctx *ctx, const jlog_id *id, jlog_message *m) {
   }
   return -1;
 }
+int jlog_ctx_bulk_read_messages(jlog_ctx *ctx, const jlog_id *id, const int count, jlog_message *m) {
```

Comment:
then this is a huge volume of code duplication... which makes maintenance more painful next time there's a bug in either routine that might affect both. It looked like this function was copied from the existing `read_message`, but with some looping logic for the bulk read op? Then it should be no risk to make the old routine use this function, assuming the tests pass?